### PR TITLE
UTM tags - add a report for UTM tags

### PR DESCRIPTION
**Pull request by Maxell92**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | -
| BC breaks? | -
| Deprecations? | -

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR adds a possibility to create a report on UTM tags with lead and company fields.

#### Steps to test this PR:
1. Checkout this PR a clear the cache.
2. Create a new report with `UTM codes` as data source
3. Try create a few reports with different column's combination, filters and group by options to test everything is working together.
4. Try to add every possible column / filter/ group by to be sure SQL is working.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.

### CODE REVIEW

**Comment by isleshocky77** [comment]
I just ran into an issue where I was unable to filter page hits based on a UTM tag of that hit; I went to see if anyone else was talking about UTM tags, and this PR popped up from yesterday. Pulled it and have a few comments.

1. It looks like Campaign is spelled incorrectly (i.e. UTM campaing)
2. It didn't actually work for me as I would expect. (e.g. I have a Contact is showing in their timeline a page hit with a "UTM Source" property, when I tried doing a report showing all Contacts from the UTM Tag data source with all UTM properties, the report was completely blank. I did not use filters or group bys.
3. This currently looks to only allow you to display UTM information by Contact. What I was looking for and hopefully isn't too big of a leap for this ticket since you have UTMTagSubscriber is to be able to report on Page Hits by UTM Tag. Specifically I'm doing reporting on Page hits grouped by Page url with a count. I would like to be able to filter those base on where the utm_source is from. I would think this would be helpful to other people as well.

**Comment by Maxell92** [comment]
Hello @isleshocky77,

1) Thanks, I will update this.
2) Can you take a look to the database, what is in the `lead_utmtags`? Report should match info from this table.
3) I am not sure how to do it. there is no connection to a page from the `lead_utmtags` table. Can you suggest how this implementation should look like? I think this is not possible. But you can send a hint / PR if you do.

**Comment by isleshocky77** [comment]
@maxell92

It looks like the data for utm for a page hit is stored in page_hits.query as json.

**Comment by javjim-mautic** [comment]
Added a group by and got 500 error.

imagen


 Logs show this

`[2018-04-26 05:57:52] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\DriverException: "An exception occurred while executing 'SELECT utm.utm_campaign AS utm_campaign, utm.utm_content AS utm_content, utm.utm_medium AS utm_medium, utm.utm_term AS utm_term, utm.utm_source AS utm_source, l.email AS email, COUNT(l.email) AS 'COUNT l.email' FROM mautic_lead_utmtags utm LEFT JOIN mautic_leads l ON l.id = utm.lead_id GROUP BY utm.utm_campaign':  SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mautician.utm.utm_content' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by" at C:\wamp64\www\mautician\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\AbstractMySQLDriver.php line 115 {"exception":"[object] (Doctrine\\DBAL\\Exception\\DriverException(code: 0): An exception occurred while executing 'SELECT utm.utm_campaign AS utm_campaign, utm.utm_content AS utm_content, utm.utm_medium AS utm_medium, utm.utm_term AS utm_term, utm.utm_source AS utm_source, l.email AS email, COUNT(l.email) AS 'COUNT l.email' FROM mautic_lead_utmtags utm LEFT JOIN mautic_leads l ON l.id = utm.lead_id GROUP BY utm.utm_campaign':\n\nSQLSTATE[42000]: Syntax error or access violation: 1055 Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mautician.utm.utm_content' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by at C:\\wamp64\\www\\mautician\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\AbstractMySQLDriver.php:115, Doctrine\\DBAL\\Driver\\PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mautician.utm.utm_content' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by at C:\\wamp64\\www\\mautician\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\PDOConnection.php:106, PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mautician.utm.utm_content' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by at C:\\wamp64\\www\\mautician\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\PDOConnection.php:104)"} []`

**Comment by Maxell92** [comment]
This is caused by your MySQL configuration - `sql_mode=only_full_group_by`. Mautic cannot work with this settings. You need to disable it, otherwise almost every report with `GROUP BY` will not work.
You can read more here: https://stackoverflow.com/questions/23921117/disable-only-full-group-by

File: app/bundles/LeadBundle/Tests/EventListener/ReportUtmTagSubscriberTest.php [hunk 1]
```diff
@@ -0,0 +1,230 @@
+<?php
+
+/*
+ * @copyright   2017 Mautic Contributors. All rights reserved
```
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/EventListener/ReportUtmTagSubscriberTest.php]: Should be 2018

**Comment by Maxell92** [comment]
I have fixed annotation + rebase branch on staging. Thanks.

**Pull request merged**